python/demo/demo_tnt-elements.py

```diff
@@ -0,0 +1,313 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.13.6
+# ---
+
+# # Creating TNT elements using Basix's custom element interface
+#
+# This demo ({download}`demo_tnt-elements.py`) illustrates how to:
+#
+# - Define custom finite elements using Basix
+#
+# We begin this demo by importing everything we require.
+
+# +
+import numpy as np
+
+from dolfinx import fem, mesh
+from ufl import dx, div, grad, inner, TestFunction, TrialFunction, sin, cos, SpatialCoordinate
+
+from mpi4py import MPI
+from petsc4py import PETSc
+
+import matplotlib.pylab as plt
+
+import basix
+import basix.ufl_wrapper
+# -
+
+# ## Defining a degree 1 TNT element
+# Basix supports a range of finite elements, but there are many other possible elements that users may
+# want to use. In this demo, we look at how Basix's custom element interface can be used to define
+# elements. More detailed information about the inputs needed to create a custom element can be found
+# in [the Basix documentation](https://docs.fenicsproject.org/basix/main/python/demo/demo_custom_element.py.html).
+#
+# As an example, we will define [tiniest tensor (TNT)](https://defelement.com/elements/tnt.html) elements on a
+# quadrilateral, as defined in [Commuting diagrams for the TNT elements on cubes
+# (Cockburn, Qiu, 2014)](https://doi.org/10.1090/S0025-5718-2013-02729-9). We begin by defining a degree 1
+# element.
+#
+# ### The polynomial set
+# We begin by defining a basis of the polynomial space that this element spans. This is defined in terms
+# of the orthogonal Legendre polynomials on the cell. For a degree 1 TNT element, the polynomial set contains
+# the polynomials $1$, $y$, $y^2$, $x$, $xy$, $xy^2$, $x^2$, and $x^2y$. These are the first 8 polynomials
+# in the degree 2 set of polynomials on a quadrilateral, so we create a 8 by 9 (number of dofs by number of
+# polynomials in the degree 2 set) matrix with an 8 by 8 identity in the first 8 columns. The order in which
+# polynomials appear in the polynomial sets for each cell are documented in [the Basix
+# documentation](https://docs.fenicsproject.org/basix/main/polyset-order.html).
+
+# +
+wcoeffs = np.zeros((8, 9))
+
+wcoeffs[0, 0] = 1
+wcoeffs[1, 1] = 1
+wcoeffs[2, 2] = 1
+wcoeffs[3, 3] = 1
+wcoeffs[4, 4] = 1
+wcoeffs[5, 5] = 1
+wcoeffs[6, 6] = 1
+wcoeffs[7, 7] = 1
```

Comment:
This seems a bit verbose, couldn't you just say: `wcoeffs[:, :-1] = np.eye(8)` ?

Reply:
Or even `wcoeffs = np.eye(8, 9)` ?